### PR TITLE
do not require nose for testing

### PR DESCRIPTION
**Pull request by pgajdos**
`nose` is deprecated and will not work in future python versions.

### CODE REVIEW

**Review by Zsailer — APPROVED** [review]

File: notebook/tests/test_paths.py [hunk 1]
```diff
@@ -16,7 +15,9 @@ def test_path_regex():
         '/x/foo/bar',
         '/x/foo/bar.txt',
     ):
-        assert_regex(path, path_pat)
+        print(type(path))
+        print(type(path_pat))
```
Comment (kevin-bates) [inline notebook/tests/test_paths.py:19]: We should probably remove these `print` statements.

**Review by kevin-bates — COMMENTED** [review]
These changes look good - thank you!  Only had the one comment regarding the print statements.

**Review by kevin-bates — APPROVED** [review]
Thanks @pgajdos.

**Comment by hroncok** [comment]
Note that nose appears still required via https://github.com/ipython/ipython_genutils/blob/fbcb174d06d6df1f8633014dd80fad8c45533c2e/ipython_genutils/testing/decorators.py#L162

**Comment by kevin-bates** [comment]
Good catch @hroncok - thanks.  Looks like the updates to `notebook/tests/test_notebookapp_integration.py` were missed.  Any idea how `nose` is getting into the CI?  I'm not seeing anything immediately, but it's definitely present in the `pip freeze` output.

At any rate, PR forthcoming to address the remaining references that pull in `nose`.

**Comment by hroncok** [comment]
AFAIK nose comes pre-installed in Travis Python environments (I guess historically this was a good idea, not sure).

**Pull request merged**